sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GatewayAddressCache.java

```diff
@@ -817,52 +833,75 @@ private static AddressInformation toAddressInformation(Address address) {
         return new AddressInformation(true, address.isPrimary(), address.getPhyicalUri(), address.getProtocolScheme());
     }
 
-    public Mono<Void> openAsync(
+    public Flux<OpenConnectionResponse> openConnectionsAndInitCaches(
             DocumentCollection collection,
             List<PartitionKeyRangeIdentity> partitionKeyRangeIdentities) {
+
+        checkNotNull(collection, "Argument 'collection' should not be null");
+        checkNotNull(partitionKeyRangeIdentities, "Argument 'partitionKeyRangeIdentities' should not be null");
+
         if (logger.isDebugEnabled()) {
-            logger.debug("openAsync collection: {}, partitionKeyRangeIdentities: {}", collection, JavaStreamUtils.toString(partitionKeyRangeIdentities, ","));
+            logger.debug(
+                    "openConnectionsAndInitCaches collection: {}, partitionKeyRangeIdentities: {}",
+                    collection.getResourceId(),
+                    JavaStreamUtils.toString(partitionKeyRangeIdentities, ","));
         }
+
         List<Flux<List<Address>>> tasks = new ArrayList<>();
         int batchSize = GatewayAddressCache.DefaultBatchSize;
```

Comment:
The default batchSize is 50, and for addressRefresh request capped at 5s. Will this be too aggressive?

Reply:
No - sounds right